javabuilder-authorizer/http_authorizer_function.rb

```diff
@@ -19,45 +20,29 @@ def lambda_handler(event:, context:)
   jwt_token = event['queryStringParameters']['Authorization']
   route_arn = event['routeArn']
 
-  decoded_token = JwtHelper.decode_token(jwt_token, origin)
+  standardized_origin = JwtHelper.get_standardized_origin(origin)
+  decoded_token = JwtHelper.decode_token(jwt_token, standardized_origin)
   return JwtHelper.generate_deny(route_arn) unless decoded_token
 
   token_payload = decoded_token[0]
-  token_status = get_token_status(context, token_payload['sid'])
+  region = get_region(context)
+  token_status = get_token_status(token_payload, standardized_origin, region)
   return JwtHelper.generate_deny(route_arn) unless token_status == TokenStatus::VALID_HTTP
 
   JwtHelper.generate_allow(route_arn, token_payload)
 end
 
-def get_token_status(context, sid)
-  client = Aws::DynamoDB::Client.new(region: get_region(context))
-
-  begin
-    ttl = Time.now.to_i + TOKEN_RECORD_TTL_SECONDS
-    client.put_item(
-      table_name: ENV['token_status_table'],
-      item: {
-        token_id: sid,
-        ttl: ttl
-      },
-      condition_expression: 'attribute_not_exists(token_id)'
-    )
-  rescue Aws::DynamoDB::Errors::ConditionalCheckFailedException
-    puts "TOKEN VALIDATION ERROR: #{TokenStatus::ALREADY_EXISTS} token_id: #{sid}"
-    # return TokenStatus::ALREADY_EXISTS
-    return TokenStatus::VALID_HTTP
-  end
-
-  # Placeholder for when we'll actually vet each token
-  # to confirm that it should not be throttled.
-  client.update_item(
-    table_name: ENV['token_status_table'],
-    key: {token_id: sid},
-    update_expression: 'SET vetted = :v',
-    expression_attribute_values: {':v': true}
-  )
-
-  TokenStatus::VALID_HTTP
+def get_token_status(token_payload, origin, region)
+  validator = TokenValidator.new(token_payload, origin, region)
+
+  return validator.error_message unless validator.log_token
+  return validator.error_message if validator.user_blocked?
+  return validator.error_message if validator.teachers_blocked?
+  return validator.error_message if validator.user_over_hourly_limit?
+  return validator.error_message if validator.user_over_daily_limit?
+  return validator.error_message if validator.teachers_over_hourly_limit?
+  validator.log_requests
+  return validator.mark_token_as_vetted
```

Comment:
nit: should this be moved to `TokenValidator` if this is going to be shared code?

Also, just a thought - I feel a little iffy on the structure of this logic. We have an assumption that the various check methods (user_blocked, etc), will internally set the @status variable that gets returned by error_message. This is internal code so it's not a huge deal, but in theory if one of these methods forgets to set the @status variable we'd be returning the wrong status value here and it may be a little tricky to nail down why. Could we just return statuses directly instead of setting them and reading them back in error_message? Something like this?

```
return TokenStatus::USER_BLOCKED if validator.user_blocked?
return TokenStatus::TEACHERS_BLOCKED if validator.teachers_blocked?
...
validator.mark_token_as_vetted
return TOKEN_STATUS::VALID_HTTP
```

Having this block inside TokenValidator may also help clean it up a bit since we'd be able to call the various checking functions directly (or even make them private)

Reply:
haha yeah a couple thoughts:
- I had it set up something like what you're describing originally (return statuses directly depending on result of each function call). I thought it'd be better to explicitly link the status and the function call that determines that status by moving the status internally, but yeah now I see how that could be confusing.
- re: "move to `TokenValidator` if this is going to be shared code", do you mean just moving this whole set of code into `TokenValidator`, or something else? I was thinking I could just add something like `validator.validate` that does all of this and just returns the validation status.

Reply:
Do you prefer something like this (no @status variable):

```
  def validate
    return error_message(ALREADY_EXISTS) unless log_token
    return error_message(USER_BLOCKED) if user_blocked?
    return error_message(TEACHERS_BLOCKED) if teachers_blocked?
    return error_message(USER_OVER_HOURLY_LIMIT) if user_over_hourly_limit?
    return error_message(USER_OVER_DAILY_LIMIT) if user_over_daily_limit?
    return error_message(TEACHERS_OVER_HOURLY_LIMIT) if teachers_over_hourly_limit?
    log_requests
    mark_token_as_vetted
    VALID_HTTP
  end
```

Reply:
Yeah this looks good! Yeah my concern was just that if we add another check but forget to set the @status variable in that function, we'd end up with a weird state

Reply:
Makes sense, give it another look when you have a chance!